client/js/shout.js

```diff
@@ -197,6 +197,23 @@ $(function() {
 				data.msg
 			]);
 
+		chan.find(".messages .text:not(.processed)")
+			.each(function() {
+				var html = $(this).html();
+				$(this)
+					.addClass("processed")
+					.html(html.replace(/(\W)(#[\w\d-.]{0,200})/ig,
```

Comment:
I searched for the specification when creating the regular expression and came up with this:

>  Channels names are strings (beginning with a '&' or '#' character) of
>  length up to 200 characters.


https://tools.ietf.org/html/rfc1459#section-1.3